autogluon/utils/tabular/ml/models/tabular_nn/tabular_nn_model.py

```diff
@@ -345,7 +348,10 @@ def train_net(self, train_dataset, test_dataset=None,
                     logger.log(15, "Epoch %s.  Train loss: %s" % (e, train_loss.asscalar()))
             if self.summary_writer is not None:
                 self.summary_writer.add_scalar(tag='train_loss', value=train_loss.asscalar(), global_step=e)  # TODO: do we want to keep mxboard support?
-            if e - best_val_epoch > self.params['epochs_wo_improve']:
+            if reporter is not None:
+                if test_dataset is not None and (not np.isnan(val_metric)):  # TODO: This might work without the if statement
```

Comment:
The `np.nan()` check is just here because the current scheduler/reporter will break if it records `nan`?
It's unclear to me what would be the best behavior in this case, but I think ignoring `nan` is a good idea for now.  However, we probably want to add a TODO here in case we later add a unified way to handle `nan` in the reporters.

Reply:
Uncertain if np.nan/None breaks it, but until reporter/scheduler is refactored I don't want to risk it.

Reply:
Added TODO